### PR TITLE
Fixing proxying for tenant owners owning multiple tenants

### LINKED ISSUE #297: Unable to proxy resources for users, who are owners of multiple tenants
# Bug description

If the user owns multiple tenants, capsule proxy won't be able to proxy get and list requests for storageclasses, ingressclasses, priorityclasses, runtimeclasses.

# How to reproduce

Create 2 tenants owned by user `alice` and one tenant owned by user `bob`. For each tenant create `storageclass`, `ingressclass`, `priorityclass`, `runtimeclass` - https://gist.github.com/MaxFedotov/d814e41a16ac2f9301058103e277216a

As `bob`, try to `list` all objects:
```bash
kubectl --kubeconfig=hack/bob.kubeconfig get storageclasses 
NAME   PROVISIONER                 RECLAIMPOLICY   VOLUMEBINDINGMODE   ALLOWVOLUMEEXPANSION   AGE
bob    csi-image.warm-metal.tech   Delete          Immediate           false                  9m41s


kubectl --kubeconfig=hack/bob.kubeconfig get ingressclasses
NAME   CONTROLLER    PARAMETERS   AGE
bob    acme.io/bob   <none>       9m13s

kubectl --kubeconfig=hack/bob.kubeconfig get runtimeclasses
NAME   HANDLER           AGE
bob    myconfiguration   6m7s

kubectl --kubeconfig=hack/bob.kubeconfig get priorityclasses
NAME   VALUE     GLOBAL-DEFAULT   AGE
bob    1000000   false            7m50s
```
Everything will succeed.

Now try the same as `alice`:
```bash
kubectl --kubeconfig=hack/alice.kubeconfig get storageclasses
No resources found

kubectl --kubeconfig=hack/alice.kubeconfig get ingressclasses
No resources found

kubectl --kubeconfig=hack/alice.kubeconfig get runtimeclasses 
No resources found

kubectl --kubeconfig=hack/alice.kubeconfig get priorityclasses
No resources found
```

According to logs, the following `label selector` will be generated for `alice` requests:
```
{"level":"Level(-4)","ts":"2023-05-22T22:45:16.875Z","logger":"proxy","msg":"labelSelector added","selector":"capsule.clastix.io/tenant=oil,capsule.clastix.io/tenant=gas"}
```

If we try to do request directly to kube-api as admin using this selector, we can see that no object will be returned:
```bash
kubectl get ingressclasses -l capsule.clastix.io/tenant=oil,capsule.clastix.io/tenant=gas
No resources found
```

As `bob`, try to `get` specific objects:
```bash
kubectl --kubeconfig=hack/bob.kubeconfig get storageclasses bob
NAME   PROVISIONER                 RECLAIMPOLICY   VOLUMEBINDINGMODE   ALLOWVOLUMEEXPANSION   AGE
bob    csi-image.warm-metal.tech   Delete          Immediate           false                  18m

kubectl --kubeconfig=hack/bob.kubeconfig get ingressclasses bob
NAME   CONTROLLER    PARAMETERS   AGE
bob    acme.io/bob   <none>       18m

kubectl --kubeconfig=hack/bob.kubeconfig get runtimeclasses bob
NAME   HANDLER           AGE
bob    myconfiguration   15m

 kubectl --kubeconfig=hack/bob.kubeconfig get priorityclasses bob
NAME   VALUE     GLOBAL-DEFAULT   AGE
bob    1000000   false            17m
```

Now try the same as `alice`:
```bash
kubectl --kubeconfig=hack/alice.kubeconfig get storageclasses oil
Error from server (NotFound): storageclasses "oil" not found

kubectl --kubeconfig=hack/alice.kubeconfig get ingressclasses oil
Error from server (NotFound): ingressclasses.networking.k8s.io "oil" not found

kubectl --kubeconfig=hack/alice.kubeconfig get runtimeclasses oil
Error from server (NotFound): runtimeclasses "oil" not found

kubectl --kubeconfig=hack/alice.kubeconfig get priorityclasses oil
Error from server (NotFound): priorityclasses "oil" not found
```

# Expected behavior

`alice` should be able to get and list all objects.

# Additional context

- Capsule-Proxy version: 5912175500d91e33551e029867c8676a51272b6b
- Kubernetes version: 1.22

### PR DESCRIPTION
**Pull request by MaxFedotov**
this PR closes #297.

### CODE REVIEW

**Review by prometherion — CHANGES_REQUESTED** [review]
I see a lot of changes in this MR, such as deprecating the support of the `v1beta1` group.

I'd ask you to create smaller changes at the commit level, and possibly, open the discussion for these drops of support in a separate MR.

I'm feeling good about the changes introduced on the `HandleListSelector`, let's go one by one to simplify the review, too.

**Comment by MaxFedotov** [comment]
@prometherion, WDYT if I split this PR into 3 parts:
1) deprecating ingress v1beta1
2) refactoring error handling
3) fix for list\get

File: internal/modules/metric/list.go [hunk 1]
```diff
@@ -22,10 +22,18 @@ import (
...
 	client client.Reader
 	log    logr.Logger
+	gk     schema.GroupKind
 }
 
 func List(client client.Reader) modules.Module {
-	return &list{client: client, log: ctrl.Log.WithName("metric_list")}
+	return &list{
+		client: client,
+		log:    ctrl.Log.WithName("metrics_get"),
```
[the remark this answers is not in the document]
Comment (MaxFedotov) [reply internal/modules/metric/list.go]: done

**Comment by MaxFedotov** [comment]
@prometherion split the commit into 2 separate and rollbacked all changes related to ingressclasses v1beta1 deprication

**Pull request merged**